### PR TITLE
[IMP]sale_order_general_discount_triple:add flag in context to check …

### PR DESCRIPTION
**Pull request by odooNextev**
…if discount already exist on sale_order_general_discount

### CODE REVIEW

**Comment by odooNextev** [comment]
I added flag in context to check if a discount already exist and avoid to add another general discount on sale_order_general_discount create method. 
https://github.com/OCA/sale-workflow/pull/2666

File: sale_order_general_discount_triple/models/sale_order_line.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ def create(self, vals):
                 "sale_order_general_discount_triple.general_discount", "discount"
             )
         )
+        flag = False
         if general_discount:
             vals[general_discount] = sale_order.general_discount
-        return super().create(vals)
+            flag = True
+        return super(SaleOrderLine, self.with_context(flag=flag)).create(vals)
```
Comment (HekkiMelody) [inline sale_order_general_discount_triple/models/sale_order_line.py:56]:
```suggestion
            self = self.with_context(no_propagate_general_discount=True)
        return super(SaleOrderLine, self).create(vals)
```

File: sale_order_general_discount_triple/models/sale_order_line.py [hunk 2]
```diff
@@ -49,6 +49,8 @@ def create(self, vals):
                 "sale_order_general_discount_triple.general_discount", "discount"
             )
         )
+        flag = False
```
Comment (HekkiMelody) [inline sale_order_general_discount_triple/models/sale_order_line.py:52]:
```suggestion

```

**Review by HekkiMelody — CHANGES_REQUESTED** [review]
It's more or less okay but a name that is more descriptive than "flag" would really help

**Comment by odooNextev** [comment]
superseeded https://github.com/OCA/sale-workflow/pull/2666

**Pull request closed without merging**